astropy/modeling/math_functions.py

```diff
@@ -0,0 +1,67 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import warnings
+import numpy as np
+
+from astropy.modeling.core import Model
+
+
+trig_ufuncs = ["sin", "cos", "tan", "arcsin", "arccos", "arctan", "arctan2",
+               "hypot", "sinh", "cosh", "tanh", "arcsinh", "arccosh",
+               "arctanh", "deg2rad", "rad2deg"]
+
+
+math_ops = ["add", "subtract", "multiply", "divide", "logaddexp", "logaddexp2",
+            "true_divide", "floor_divide", "negative", "positive", "power",
+            "remainder", "mod", "fmod", "divmod", "absolute", "fabs", "rint",
+            "exp", "exp2", "log", "log2", "log10", "expm1", "log1p", "sqrt",
+            "square", "cbrt", "reciprocal"]
+
+
+supported_ufuncs = trig_ufuncs + math_ops
+
+
+class _NPUfuncModel(Model):
+    _is_dynamic = True
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def evaluate(self, x):
```

Comment:
Does this ever get called since the factory function below supplies a new definition? If not, should the warning go in the init instead?

Reply:
Ah, right, I moved the definition to the factory function and forgot to delete this one.